### DISCUSSION TITLE
Smarter value expansion in secret references

### DISCUSSION THREAD

**Discussion by paolostyle** [Infisical/infisical#6309] [Feature Request]
### Feature description
Currently I cannot reference a secret from the same environment that is located in a different folder without specifying the environment. For example, let's say I have a `dev` and `prod` environments, and both of them contain a folder `common` containing a secret MY_SECRET and I want to reference it in a secret located in a folder `project`. To do so, the interpolation has to look like this: `${dev.common.MY_SECRET}` for `dev` and `${prod.common.MY_SECRET}` for `prod`. `${common.MY_SECRET}` throws an error "`common` environment does not exist".

It's quite annoying because I'm not able to add such secret from the Overview page which adds the same secret to multiple environments at the same time, or rather I can, but I need to adjust all of them manually. I would much rather have both of them be `${common.MY_SECRET}`, and by default assume that it's coming from the same environment. If there was a `common` environment and `common` folder, that would indeed be ambiguous and would require specifying the environment. I think situations like this would be very unlikely, but backwards compatibility would definitely have to be considered.

### Why would it be useful?
I think adding secrets through the Overview page is super powerful and this would immensely reduce the tediousness of the current behavior.

**Comment by mmueller-rs** [comment]
+1
That's exactly what I'd like to do as well.

**Comment by akhilmhdh** [comment]
Moving this to [feature request discussion](https://github.com/Infisical/infisical/issues/5172).
